### ISSUE TITLE
The handle-done-callback rule cannot handle TypeScript this parameter

### ISSUE THREAD

**Issue by raymondfeng** [lo1tuma/eslint-plugin-mocha#270]
The issue can be reproduced with the following TypeScript code:

```ts
before(async function setupApplication(this: Mocha.Context) {
    this.timeout(6000);
}
```

Please note [this parameter](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters) is used. 

The `handle-done-callback` rule marks it as a violation incorrectly.

It's also interesting to note that the function is `async`.

**Comment by lo1tuma** [comment]
Which parser are you using? I think this should be rather fixed in `@typescript-eslint/parser`.

**Comment by raymondfeng** [comment]
I use `@typescript-eslint/parser`

**Comment by raymondfeng** [comment]
@lo1tuma Do you recommend we open an issue against `@typescript-eslint/parser`?

**Comment by Ulrikop** [comment]
I run into the same problem. Is there any other solution than deactivating the rule for TypeScript files?
